### PR TITLE
iox-#1755 Port to new macro

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

This PR ports the code to the new log macro with lazy evaluation and removes the old macro.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Relates to #1755

### CODE REVIEW

**Comment by elBoberido** [comment]
@FerdinandSpitzschnueffler @mossmaurice @dkroenke @MatthiasKillat @elfenpiff @budrus feel free to add yourself as reviewer

File: iceoryx_hoofs/source/posix_wrapper/named_semaphore.cpp [hunk 1]
```diff
@@ -40,11 +40,12 @@ static expected<void, SemaphoreError> unlink(const NamedSemaphore::Name_t& name)
...
         case EACCES:
-            IOX_LOG(ERROR) << "You don't have permission to remove the semaphore \"" << name << "\"";
+            IOX_LOG(ERROR, "You don't have permission to remove the semaphore \"" << name << '"');
             return err(SemaphoreError::PERMISSION_DENIED);
         default:
-            IOX_LOG(ERROR) << "This should never happen. An unknown error occurred while creating the semaphore \""
-                           << name << "\".";
+            IOX_LOG(ERROR,
+                    "This should never happen. An unknown error occurred while creating the semaphore \"" << name
+                                                                                                          << '"');
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_hoofs/source/posix_wrapper/named_semaphore.cpp:48]: You forgot the period :stuck_out_tongue_winking_eye:

**Review by elfenpiff — APPROVED** [review]

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

File: iceoryx_hoofs/source/posix_wrapper/named_semaphore.cpp (hunk 1 above)
Comment (elBoberido) [reply iceoryx_hoofs/source/posix_wrapper/named_semaphore.cpp:48]: Will fix it in the follow up PR :)

Comment (FerdinandSpitzschnueffler) [reply iceoryx_hoofs/source/posix_wrapper/named_semaphore.cpp:48]: That's not necessary, IMO.

Comment (elBoberido) [reply iceoryx_hoofs/source/posix_wrapper/named_semaphore.cpp:48]: Yeah, I also just noticed that there already were quite a few inconsistencies with the period

**Pull request merged**